node/engine/store/memstore.go

```diff
@@ -18,13 +19,18 @@ import (
 	"github.com/statechannels/go-nitro/types"
 )
 
+type blockData struct {
+	lastBlockNumSeen uint64
+	mu               sync.Mutex
+}
+
 type MemStore struct {
 	objectives         safesync.Map[[]byte]
 	channels           safesync.Map[[]byte]
 	consensusChannels  safesync.Map[[]byte]
 	channelToObjective safesync.Map[protocols.ObjectiveId]
 	vouchers           safesync.Map[[]byte]
-	lastBlockNumSeen   uint64
+	blocks             blockData
```

Comment:
Is it more clear to name this `lastBlockSeen` or something like that? `blocks` is somewhat ambiguous...

Reply:
Yea I think that's a better name. Changed in https://github.com/statechannels/go-nitro/pull/1650/commits/e854ac45f9cddbe242ec43ba5490460e6e36c139